pkgs/applications/networking/cluster/terraform-providers/update-provider

```diff
@@ -47,31 +96,41 @@ prefetch_github() {
   nix-prefetch-url --unpack "https://github.com/$owner/$repo/archive/$rev.tar.gz"
 }
 
+version="$(jq -r '.version' <<< "$registry_response")"
+if [[ "$old_version" = "$version" && "$force" != 1 ]]; then
+  echo "$provider_name is already at version $version"
+  exit
+fi
+update_attr version "$version"
+
 provider_source_url="$(jq -r '.source' <<< "$registry_response")"
 
 org="$(echo "$provider_source_url" | cut -d '/' -f 4)"
+update_attr owner "$org"
 repo="$(echo "$provider_source_url" | cut -d '/' -f 5)"
+update_attr repo "$repo"
 rev="$(jq -r '.tag' <<< "$registry_response")"
-
+update_attr rev "$rev"
 sha256=$(prefetch_github "$org" "$repo" "$rev")
+update_attr sha256 "$sha256"
 
-version="$(jq -r '.version' <<< "$registry_response")"
-
-updated_provider="$(mktemp)"
-cat <<EOF >> "$updated_provider"
-{
-  "$provider_name": {
-    "owner": "$org",
-    "repo": "$repo",
-    "rev": "$rev",
-    "sha256": "$sha256",
-    "version": "$version",
-    "provider-source-address": "$provider_source_address"
-  }
-}
-EOF
+repo_root=$(git rev-parse --show-toplevel)
 
-original_provider_list="$(mktemp)"
-cat providers.json > "$original_provider_list"
+if [[ $old_vendor_sha256 != "null" || "$vendor" = 1 ]]; then
+  update_attr vendorSha256 "0000000000000000000000000000000000000000000000000000000000000000"
+  # Hackish way to find out the desired sha256. First build, then extract the
+  # error message from the logs.
+  set +e
+  logs=$(nix-build "$repo_root" -A "terraform-providers.$provider_name.go-modules" 2>&1)
+  set -e
+  [[ $logs =~ got:\ +([^\ ]+) ]]
+  vendorSha256=${BASH_REMATCH[1]}
+  # Deal with nix unstable
+  if [[ $vendorSha256 = sha256-* ]]; then
```

Comment:
```suggestion
  if [[ $vendorSha256 == sha256-* ]]; then
```

Reply:
Does it change the semantic?

Reply:
shellcheck doesn't have an opinion on this